controllers/snapshot/snapshot_adapter.go

```diff
@@ -154,11 +155,20 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (reconciler.OperationResul
 		for _, component := range a.snapshot.Spec.Components {
 			if component.Name == a.component.Name {
 				a.component.Spec.ContainerImage = component.ContainerImage
-				err := a.client.Patch(a.context, a.component, patch)
-				if err != nil {
-					a.logger.Error(err, "Failed to update Global Candidate for the Component",
+				err1 := a.client.Patch(a.context, a.component, patch)
```

Comment:
Just use "err" instead of "err1" (and the following lines), then change line 166 to "err = a.client.Status().Patch(a.context, a.component, patch)" will be ok.

Reply:
Done, thank you!